x-pack/plugins/apm/server/routes/settings/anomaly_detection.ts

```diff
@@ -44,11 +79,30 @@ export const createAnomalyDetectionJobsRoute = createRoute(() => ({
   handler: async ({ context, request }) => {
     const { environments } = context.params.body;
     const setup = await setupRequest(context, request);
-    return await createAnomalyDetectionJobs(
-      setup,
-      environments,
-      context.logger
-    );
+
+    try {
+      await createAnomalyDetectionJobs(setup, environments, context.logger);
+    } catch (e) {
+      // InsufficientMLCapabilities
+      if (e instanceof InsufficientMLCapabilities) {
+        return {
+          errorCode: ErrorCode.MISSING_READ_PRIVILEGES,
+        };
+      }
+
+      // AnomalyDetectionError
+      if (e instanceof AnomalyDetectionError) {
+        return {
+          errorCode: e.code,
+        };
+      }
+
+      // unexpected error
+      context.logger.warn(e.message);
+      return {
+        errorCode: ErrorCode.UNEXPECTED,
+      };
+    }
```

Comment:
And based on the changes above:
```suggestion
      const mlErrorCode = getMLErrorCode(e);
      return {
         errorCode: mlErrorCode
      }
```